### ISSUE TITLE
'ICV check failed!' being reported with valid WEP key

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#357]
**Reported by andy a t burningimage net on 24 Nov 2007 16:57 UTC**

I'm trying to monitor all my encrypted wireless traffic.

The command I'm running is:
airtun-ng -a 00:40:10:10:00:03 -w <26 HEX digits> ath1

I'm convinced the WEP key is correct (it works in managed mode), but whenever I send some traffic over the wireless, airtun-ng reports "ICV check failed!".

Not sure if it's got anything to do with it, but a chopchop attack also doesn't work, I get the "not properly discarding frames" error.  Fragmentation attacks and injection work fine.  Using patched madwifi-ng drivers.

Same story with 0.9.1.

**Comment by aircrack-ng** [comment]
**Modified by misterx on 24 Nov 2007 16:57 UTC**

**Comment by aircrack-ng** [comment]
**Comment by anonymous on 24 Nov 2007 16:57 UTC**

There is a problem using airtun-ng with clients that use the QoS data frames. If the QoS bit is set, there are two extra octets of QoS information that need to be allowed for when computing the offset to the encrypted payload.

The following fix in airtun-ng (around line 565) resolves this issue:

 ```
    z = ( ( packet[& 3 ) != 3 ) ? 24 : 30;
    /* If the QoS bit is set add two to the offset */
    if( packet[0](1]) & 0x80 ) z += 2;
 ```

**Comment by aircrack-ng** [comment]
**Comment by misterx on 24 Nov 2007 16:57 UTC**

Fixed in aircrack-ng/aircrack-ng@0f5b7a6625f3af409887fa9e1e7b9f76eca2db7d
